### PR TITLE
Fixup test_list_volumes.py failure

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes intermittent test failures due left over volumes from previous test runs by waiting for their cleanup.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: test/integration/smoke/test_list_volumes.py [hunk 1]
```diff
@@ -140,6 +140,26 @@ def setUpClass(cls):
     def tearDownClass(cls):
         super(TestListVolumes, cls).tearDownClass()
 
+    @classmethod
+    def wait_for_volume_cleanup(cls):
+        """Clean up volumes that were left by previous tests
```
Comment (sureshanaparti) [inline test/integration/smoke/test_list_volumes.py:145]:
```suggestion
        """Wait for volumes to clean up that were left by previous tests
```

this doesn't cleanup, waiting for earlier volumes to tear down.

**Comment by sureshanaparti** [comment]
@vishesh92 can you update the description (about the issue, fix)

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code lgtm

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]

**Comment by vishesh92** [comment]
@shwstppr @sureshanaparti After a discussion with @DaanHoogland, I have made changes to wait for cleanup of volumes after every test run.

**Comment by shwstppr** [comment]
@vishesh92 cc @DaanHoogland @sureshanaparti @weizhouapache 
I do not like the idea of the listing tests failing if there are other resources present. I would prefer reverting the original PR to have stable results for the RC unless others are in line with the changes

**Comment by shwstppr** [comment]
@DaanHoogland https://github.com/apache/cloudstack/pull/8281
Yes, changes in this are fine but even with them I won't be able to successfully run some of the tests added in #8281 if my env already has some resources

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
Included in https://github.com/apache/cloudstack/pull/8410

**Pull request closed without merging**